### PR TITLE
Updated MLOS documentation

### PR DESCRIPTION
**Pull request by carolinepacheco**
Some updates:

* auto ml optimization catalog;
* deep learning workflows catalog;
* data visualization catalog;
* fixed some typos.

### CODE REVIEW

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 1]
```diff
@@ -283,24 +283,28 @@ In the following, we briefly describe the different tuners proposed by the Multi
...
+*Himmelblau_Function:* is a multi-modal function, used to test the performance of optimization algorithms.
+
+
+=== Hyperparameter Optimization
 
 The following workflows represent some machine learning and deep learning algorithms that can be optimized.
 These workflows have several common variables as in Multi_Tuners_Auto_ML. Some of the workflows are characterized
 by few additional variables.
 
+*CIFAR_10_Image_Classification:* train a simple deep CNN on the CIFAR10 images dataset using the Keras library.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:299]: train => trains

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 2]
```diff
@@ -309,43 +313,41 @@ by few additional variables.
...
 
-*Text_Generation:* train a simple Long Short-Term Memory (LSTM) to learn sequences of characters from 'The Alchemist' book. It's a novel by Brazilian author Paulo Coelho that was first published in 1988.
 
+=== Neural Architecture Search
 
-=== Objective Function Examples
+The following workflows contain a search space containing a set of possible neural networks architectures that can be used by the AutoML tuners to automatically find the best combinations of  neural architectures within the search space.
 
-The following workflows represent some mathematical functions that can be optimized by the AutoML tuners.
+*Handwritten_Digit_Classification:* train a simple deep CNN on the MNIST dataset using the Pytorch library.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:342]: trains => trains

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 3]
```diff
@@ -3882,10 +3900,23 @@ Visdom bucket provides a fast, easy and practice way to execute different workfl
...
+| String (default="visdom-server")
+| `PROXIFIED`
+| It takes by default the value of  `VISDOM_PROXYFIED` workflow variable.
+| String (default="$VISDOM_PROXYFIED")
+(default=empty)
+| `ENGINE`
+| Container engine.
+| String (default="$CONTAINER_PLATFORM")
+| `NATIVE_SCHEDULER`
+| Name of the Native Scheduler node source to use when the workflowtasks must be deployed inside a cluster such as SLURM, LSF, etc.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:3915]: workflowtasks => workflow tasks

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 4]
```diff
@@ -3915,9 +3949,16 @@ WARNING: If two workflows use the same service instance names, then, their gener
 *Task Variables:*
 |===
 | *Variable name* | *Description* | *Type*
+| `TASK_ENABLED`
+|If False, the will be ignored, it will not be executed.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:3953]: the will be ignored => the task will be ignored

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 5]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
 
 *Visdom_Real_time_Digit_Classification:* shows an example of realtime plotting using the Visdom server for training a convolutional neural network (CNN) for MNIST digit classification.
 
+*Check_Visdom_Support:* checks if the user wants (or not) start the Visdom service.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:3989]: wants (or not) start => wants (or not) to start

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 6]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
...
+
 A demo video of these workflows is available in https://www.youtube.com/user/ActiveEonCompany[ActiveEon Youtube Channel^].
 
+==== Tensorboard
+
+It provides  the visualization and tooling needed for machine learning experimentation, such as tracking and visualizing metrics such as loss and accuracy, visualizing the model graph (ops and layers), viewing histograms of weights, biases, or other tensors as they change over time, projecting embeddings to a lower dimensional space, displaying images, text, and audio data and others.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:3995]: for machine learning experimentation, such as tracking => for the machine learning experimentation such as tracking,

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 7]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
...
+
+.Tensorboard_Service_Start variables
+[cols="2,5,2"]
+|===
+| *Variable name* | *Description* | *Type*
+| `SERVICE_ID`
+| The id of the TensorBoard service.
+| String (default="Tensorboard")
+| `INSTANCE_NAME`
+| The instance name of the server to use to broadcast visualization
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:4011]: of the server to use to broadcast visualization => of the server to be used to broadcast the visualization

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 8]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
...
+| String (default=/shared/$TENSORBOARD_HOST_LOG_PATH)
+| `ENGINE`
+| Container engine.
+| String (default="$CONTAINER_PLATFORM")
+| `PROXIFIED`
+| It takes by default the value of  `TENSORBOARD_PROXYFIED` workflow variable.
+| String (default="$TENSORBOARD_PROXYFIED")
+(default=empty)
+| `NATIVE_SCHEDULER`
+| Name of the Native Scheduler node source to use when the workflowtasks must be deployed inside a cluster such as SLURM, LSF, etc.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:4024]: workflowtasks =>  workflow tasks

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 9]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
...
+| Parameters given to the native scheduler (SLURM, LSF, etc) while requesting a ProActive node used to deploy the workflow tasks.
+| String (default=empty)
+| `CONTAINER_ROOTLESS_ENABLED`
+| If True, the user will be able to run the workflow in a rootless mode.
+| (default=False)
+|===
+
+===== Tensorboard_Service_Actions
+
+*Task Overview:* Manage the life-cycle of TensorBoard PCA service. It allows to trigger three possible actions: Pause_Tensorboard, Resume_Tensorboard and Finish_Tensorboard.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:4036]: Manage => It manages

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 10]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
...
+| `ACTION`
+| The action that will be processed regarding the service status.
+| List [Pause_Tensorboard, Resume_Tensorboard and Finish_Tensorboard] (default="Finish_Tensorboard")
+|===
+
+==== Tensorboard Workflows
+
+The following workflows present some examples using TensorBoard service to visualize the results obtained while training and testing some predictive models.
+
+*Tensorboard_Realtime_CIFAR10_Training:* shows an example of real-time using Tensorboard for training a CNN using CIFAR10 database.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:4059]: shows an example of real-time ?? of real-time what?

File: src/docs/MLOS/MLOSUserGuide.adoc [hunk 11]
```diff
@@ -3945,8 +3986,83 @@ video::bQpwS2nMAZY[youtube, width=700, height=400 start=0, position=center]
...
+
+*Tensorboard_Plots_Example:* shows an example exposing the different plots available in Tensorboard.
+
+*Check_Tensorboard_Support:* checks if the user wants (or not) start the Tensorboard service.
```
Comment (dianajlailaty) [inline src/docs/MLOS/MLOSUserGuide.adoc:4063]: wants (or not) start => wants (or not) to start

**Review by dianajlailaty — CHANGES_REQUESTED** [review]

**Review by ImenBizid — APPROVED** [review]

**Review by andrewssobral — APPROVED** [review]

**Pull request merged**